[Docs] XMLRPC - Repair Syntax, Params, Descriptions and Refs

seaniepie opened the pull request:
All unified and fixed

Commit:
[Docs] XMLRPC - Repair Syntax, Params, Descriptions and Refs

peter-b commented on docs/dictionary/function/revXMLRPC_GetPath.lcdoc:
```diff
@@ -18,40 +18,41 @@ Platforms: desktop, server, mobile
-executed an XML-RPC request.
+XMLRPCdocumentID (integer):
+The number returned by the <revXMLRPC_CreateRequest> when you created the
+<XML-RPC> request, or the successful <revXMLRPC_Execute< <function> when you
```
`<revXMLRPC_Execute<` should be `<revXMLRPC_Execute>`.

peter-b commented on docs/dictionary/function/revXMLRPC_GetHost.lcdoc:
```diff
@@ -18,39 +18,40 @@ Platforms: desktop, server, mobile
+> <standalone application>, you must include this
+> <LiveCode custom library|custom library> when you create your
+> <standalone application|standalone>. In the Inclusions pane of the
+> <Standalone Application Settings> window, make sure the "XML-RPC"
```
I note that only a minority of the XML-RPC library syntax dictionary entries have this "Important" paragraph.  However, I'm not sure it's ideal to copy and paste it into every dictionary entry related to the XML-RPC library.  Do you think there's any logical place to just put the warning once, but where most people using the library are likely to see it?  Is there an overall summary page for the XML-RPC library that might be a good place to stash it?

peter-b reviewed (commented):
I just noticed a couple of issues while looking through this.  Big improvement overall, though!

seaniepie commented:
The most obvious place would be to add it into the revXMLRPC_CreateRequest as this will be the first place  to visit for any first time XMLRPC user. After that there are a few other key one like revXMLRPC_Execute, revXMLRPC_GetRequest, revXMLRPC_GetResponse, revXMLRPC_SetSocket and perhaps revXMLRPC_DeleteAllDocuments and revXMLRPC_DeleteDocument. Then the glossary for 'XML-RPC' and 'XML-RPC document'

seaniepie committed:
update - angle bracket direction mistake

peter-b commented:
@seaniepie That sounds like a sensible approach. Do you think it would be best to make sure that the inclusion note is in all the right places as part of this PR, or maybe as a separate one?

seaniepie commented:
Lets go for it. I think it would be nice to have this done _and_ dusted. With that in mind does revXMLRPC_DeleteParam need a result value? Does it also respond with the err if there is one?

I'll make and add in to this repo the changes to the glossary items.

peter-b commented:
From the [source code](https://github.com/livecode/livecode-ide/blob/5b6cb938ae716e1c691d3876ce4eee6b04bfecc8/Toolset/libraries/revxmlrpclibrary.livecodescript#L146) it appears that `revXMLRPC_DeleteParam` sets the result to an error message if it fails to delete the parameter.

seaniepie commented:
I'm making sure that a link to the XML-RPC library (library) is included in every doc too.

Commit:
update - Library notice and refs
- The *important* notice is only to be included in key functions/commands
- Link to XMLRPC library on every document
- Err result added to DeleteParam command doc

montegoulding commented:
@livecode-vulcan review ok 371d684

livecode-vulcan commented:
💙 review by @montegoulding ok 371d684c074a63485dbb95a27b32704b4cc11335

livecode-vulcan commented:
😎 test success 371d684c074a63485dbb95a27b32704b4cc11335
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/542)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/547)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/542)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/546)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/551)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/543)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/543)
